### PR TITLE
[mono] Implement missing SSE2 intrinsics

### PR DESCRIPTION
**Pull request by EgorBo**
https://github.com/dotnet/runtime/pull/33465 implemented 99% of them but a few were missing. This PR implements the missing pieces and unlocks `Sse2.IsSupported`.

@vargaz I hope I didn't step on your toe (feel free to close if you already have it locally).

### CODE REVIEW

File: src/mono/mono/mini/mini-llvm.c [hunk 1]
```diff
@@ -7109,6 +7109,12 @@ process_bb (EmitContext *ctx, MonoBasicBlock *bb)
 			values [ins->dreg] = LLVMBuildSIToFP (builder, shuffle, LLVMVectorType (LLVMDoubleType (), 2), dname);
 			break;
 		}
+		case OP_CVTSD2SD: {
+			LLVMValueRef rhs_elem = LLVMBuildExtractElement (builder, rhs, LLVMConstInt (LLVMInt32Type (), 0, FALSE), "");
+			LLVMValueRef fpext = LLVMBuildFPExt (builder, rhs_elem, LLVMDoubleType (), dname);
+			values [ins->dreg] = LLVMBuildInsertElement (builder, lhs, fpext, LLVMConstInt (LLVMInt32Type (), 0, FALSE), "");
+			break;
+		}
```
Comment (EgorBo) [inline src/mono/mono/mini/mini-llvm.c]: https://godbolt.org/z/-Y23GG

File: src/mono/mono.proj [hunk 2]
```diff
@@ -577,7 +577,7 @@
 
   <Target Name="ValidateLocalDotnet">
     <Error Condition="$([System.IO.Directory]::GetDirectories('$(LocalDotnetDir)\shared\Microsoft.NETCore.App').Length) > 1" Text="$(LocalDotnetDir) contains more than one SDK, please delete unused ones." />
-    <Error Condition="$([System.IO.Directory]::GetDirectories('$(LocalMonoDotnetDir)\shared\Microsoft.NETCore.App').Length) > 1" Text="$(LocalMonoDotnetDir) contains more than one SDK, please delete unused ones." />
+    <Error Condition="Exists('$(LocalMonoDotnetDir') AND $([System.IO.Directory]::GetDirectories('$(LocalMonoDotnetDir)\shared\Microsoft.NETCore.App').Length) > 1" Text="$(LocalMonoDotnetDir) contains more than one SDK, please delete unused ones." />
     <Error Condition="!Exists('$(LocalDotnet)')" Text="'$(LocalDotnet)' doesn't exist." />
```
Comment (vargaz) [inline src/mono/mono.proj]: Unrelated.

File: src/mono/mono/mini/simd-intrinsics-netcore.c [hunk 3]
```diff
@@ -1158,48 +1157,62 @@ emit_x86_intrinsics (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSignature
...
-				if (arg0_type == MONO_TYPE_I8 || arg0_type == MONO_TYPE_U8)
-					return emit_simd_ins_for_sig (cfg, klass, OP_SSE2_PSRLQ_IMM, -1, arg0_type, fsig, args);
-				else {
-					g_assert_not_reached ();
-					break;
-				}
-			} else {
-				// FIXME:
-				return NULL;
+		case SN_ShiftRightLogical: {
```
Comment (EgorBo) [inline src/mono/mono/mini/simd-intrinsics-netcore.c]: SN_ShiftRightLogical
SN_ShiftRightArithmetic
SN_ShiftLeftLogical
were implemented only for "imm" overloads.

**Review by vargaz — APPROVED** [review]

**Review by filipnavara — APPROVED** [review]
